pkg/cache/clusterqueue_snapshot.go

```diff
@@ -0,0 +1,62 @@
+package cache
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/metrics"
+	"sigs.k8s.io/kueue/pkg/resources"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+type ClusterQueueSnapshot struct {
```

Comment:
Did you consider creating a new package `snapshot`? So we could have `snapshot.ClusterQueue`

Reply:
I did it this way mainly to keep the diff small. If I create a new Snapshot package, I will have to move all the Snapshot functions/tests to the new files/package. Additionally, I'll have to move some additional structures (such as ResourceGroup) to a third package, to avoid any circular dependencies.

If you don't mind the much larger diff, I am happy to do it this way too. If yes, do you prefer `pkg/snapshot` or `pkg/cache/snapshot` as the directory structure for this new snapshot package?